services/harmony/app/backends/workflow-orchestration/work-item-polling.ts

```diff
@@ -224,15 +224,23 @@ export async function getWorkFromQueue(serviceID: string, reqLogger: Logger): Pr
     await queue.deleteMessage(queueItem.receipt);
     reqLogger.debug(`Deleted work item with receipt ${queueItem.receipt} from queue ${queueUrl}`);
     const item = JSON.parse(queueItem.body) as WorkItemData;
-    // get the operation for the step
-    const operationJson = await operationCache.fetch(`${item.workItem.jobID},${item.workItem.serviceID}`);
+
+    // always fetch for query-cmr service to make sure the update to operation
+    // for removing the extraArgs used for granule validation is picked up
+    const operationKey = `${item.workItem.jobID},${item.workItem.serviceID}`;
+    if (QUERY_CMR_SERVICE_REGEX.test(item.workItem.serviceID)) {
```

Comment:
Not sure I would even bother inserting into the cache here for queryCMR. I get that it makes the call on line 235 the same for all services, but I would probably just include that in an `else` and use the value returned by this fetch directly.